app/retry/retry_internal_test.go

```diff
@@ -0,0 +1,26 @@
+// Copyright © 2022-2025 Obol Labs Inc. Licensed under the terms of a Business Source License 1.1
+
+package retry
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/obolnetwork/charon/app/expbackoff"
+)
+
+func TestDelayForIteration(t *testing.T) {
```

Comment:
How long does that test take to execute?

Reply:
Instant. The function only calculates durations. It does not sleep/awaits.